src/passes/Precompute.cpp

```diff
@@ -226,16 +235,20 @@ struct Precompute
   // Precompute an expression, returning a flow, which may be a constant
   // (that we can replace the expression with if replaceExpression is set).
   Flow precomputeExpression(Expression* curr, bool replaceExpression = true) {
-    if (!canEmitConstantFor(curr->type)) {
-      return Flow(NONCONSTANT_FLOW);
-    }
+    Flow flow;
     try {
-      return PrecomputingExpressionRunner(
-               getModule(), getValues, replaceExpression)
-        .visit(curr);
+      flow =
+        PrecomputingExpressionRunner(getModule(), getValues, replaceExpression)
+          .visit(curr);
```

Comment:
It might return a breaking flow in general (nonconstant or due to something else). In that case it should not matter much what we return, but you're right that this is confusing, I'll fix it.